aws/resource_aws_wafregional_web_acl_association.go

```diff
@@ -74,21 +74,31 @@ func resourceAwsWafRegionalWebAclAssociationRead(d *schema.ResourceData, meta in
 
 	webAclId, resourceArn := resourceAwsWafRegionalWebAclAssociationParseId(d.Id())
 
-	// List all resources for Web ACL and see if we get a match
-	params := &wafregional.ListResourcesForWebACLInput{
-		WebACLId: aws.String(webAclId),
-	}
-
-	resp, err := conn.ListResourcesForWebACL(params)
-	if err != nil {
-		return err
-	}
-
-	// Find match
+	/*
+		AWS does not return all resource types on ListResourcesForWebACL, default
+		behavior is to only return ALB resources. Make call for each possible
+		resource type
+	*/
 	found := false
-	for _, listResourceArn := range resp.ResourceArns {
-		if resourceArn == *listResourceArn {
-			found = true
+	RESOURCE_TYPES := []string{"APPLICATION_LOAD_BALANCER", "API_GATEWAY"}
+	for _, resource := range RESOURCE_TYPES {
+		// List all resources for Web ACL of resource type to see if theres a match
+		params := &wafregional.ListResourcesForWebACLInput{
+			WebACLId:     aws.String(webAclId),
+			ResourceType: aws.String(resource),
+		}
+		resp, err := conn.ListResourcesForWebACL(params)
```

Comment:
Instead of using the `ListResourcesForWebACL` API call, which can be expensive for large environments, is it possible to use [`GetWebACLForResource`](https://docs.aws.amazon.com/sdk-for-go/api/service/wafregional/#WAFRegional.GetWebACLForResource)? Its seems this could perform the lookup without looping through results.